models/keyword_test.go

```diff
@@ -0,0 +1,182 @@
+package models_test
+
+import (
+	"fmt"
+	"go-crawler-challenge/models"
+	. "go-crawler-challenge/tests"
+	. "go-crawler-challenge/tests/fixtures"
+
+	"github.com/bxcodec/faker/v3"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Keyword", func() {
+	AfterEach(func() {
+		TruncateTable("keyword")
+		TruncateTable("user")
+	})
+
+	Describe("#GetAllKeyword", func() {
+		Context("given an existing keyword", func() {
+			It("returns a keyword record", func() {
+				user := FabricateUser(faker.Email(), faker.Password())
+				keyword := FabricateKeyword(faker.Word(), "https://www.google.com/search?lr=lang_en", user)
+				orderBy := []string{"created_at desc"}
+				offset := 0
+				limit := 1
+
+				keywords, err := models.GetAllKeyword(orderBy, int64(offset), int64(limit))
+				if err != nil {
+					Fail(fmt.Sprintf("Get all keywords failed: %v", err.Error()))
+				}
+
+				Expect(len(keywords)).To(BeNumerically(">", 0))
+				Expect(keywords[0].Id).To(Equal(keyword.Id))
+			})
+
+			Context("given order by `id` in ascending order", func() {
+				It("orders the first record to the top of the list", func() {
+					orderBy := []string{"id asc"}
+
+					user := FabricateUser(faker.Email(), faker.Password())
+					firstKeyword := FabricateKeyword(faker.Word(), "https://www.google.com/search?lr=lang_en", user)
+					secondKeyword := FabricateKeyword(faker.Word(), "https://www.google.com/search?lr=lang_en", user)
+					offset := 0
+					limit := 2
+
+					keywords, err := models.GetAllKeyword(orderBy, int64(offset), int64(limit))
+					if err != nil {
+						Fail(fmt.Sprintf("Get all keywords failed: %v", err.Error()))
+					}
+
+					Expect(keywords[0].Id).To(Equal(firstKeyword.Id))
+					Expect(keywords[0].Id).NotTo(Equal(secondKeyword.Id))
```

Comment:
Hits assertion is kind of convoluted why not asserting on the second record:

```suggestion
					Expect(keywords[1].Id).To(Equal(secondKeyword.Id))
```